onnxruntime/core/framework/tensor_type_and_shape.cc

```diff
@@ -151,45 +154,34 @@ ONNXTensorElementDataType MLDataTypeToOnnxRuntimeTensorElementDataType(
   return TensorDataTypeToOnnxRuntimeTensorElementDataType(prim_type->GetDataType());
 }
 
-OrtStatus* GetTensorShapeAndTypeHelper(ONNXTensorElementDataType type, const onnxruntime::TensorShape shape,
-                                       const std::vector<std::string>* dim_params, OrtTensorTypeAndShapeInfo** out) {
-  OrtTensorTypeAndShapeInfo* ret;
-  if (auto* status = OrtApis::CreateTensorTypeAndShapeInfo(&ret))
-    return status;
-  if (auto* status = OrtApis::SetTensorElementType(ret, type)) {
-    OrtApis::ReleaseTensorTypeAndShapeInfo(ret);
-    return status;
-  }
-
-  auto* status = OrtApis::SetDimensions(ret, shape.GetDims().data(), shape.GetDims().size());
-  if (status != nullptr) {
-    OrtApis::ReleaseTensorTypeAndShapeInfo(ret);
-    return status;
-  }
+OrtTensorTypeAndShapeInfo::Ptr OrtTensorTypeAndShapeInfo::GetTensorShapeAndTypeHelper(ONNXTensorElementDataType type, onnxruntime::TensorShape shape,
```

Comment:
nit: line length in a lot of places. 

https://marketplace.visualstudio.com/items?itemName=PaulHarrington.EditorGuidelinesPreview can be used to add a vertical ruler at 120 chars